cluster-autoscaler/cloudprovider/azure/azure_scale_set_test.go

```diff
@@ -158,3 +159,29 @@ func TestDebug(t *testing.T) {
 	asg.Name = "test-scale-set"
 	assert.Equal(t, asg.Debug(), "test-scale-set (5:55)")
 }
+
+func TestScaleSetNodes(t *testing.T) {
+	provider := newTestProvider(t)
+	registered := provider.azureManager.RegisterAsg(
+		newTestScaleSet(provider.azureManager, "test-asg"))
+	assert.True(t, registered)
+	assert.Equal(t, len(provider.NodeGroups()), 1)
+
+	fakeProviderID := "azure://" + fakeVirtualMachineScaleSetVMID
+	node := &apiv1.Node{
+		Spec: apiv1.NodeSpec{
+			ProviderID: fakeProviderID,
+		},
+	}
+	group, err := provider.NodeGroupForNode(node)
+	assert.NoError(t, err)
+	assert.NotNil(t, group, "Group should not be nil")
+	assert.Equal(t, group.Id(), "test-asg")
+	assert.Equal(t, group.MinSize(), 1)
+	assert.Equal(t, group.MaxSize(), 5)
+
+	instances, err := group.Nodes()
```

Comment:
Shouldn't we do this twice to make sure that the cache is working or check whether the cache got populated ?

Reply:
Good catch. Added validation for it.